docs/madr/decisions/073-spiffe-compliance.md

```diff
@@ -153,7 +157,753 @@ Future releases
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshIdentity
+metadata:
+  name: old-identity
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  selector:
+    dataplane:
+      matchLabels: {} # select all
+....
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshIdentity
+metadata:
+  name: new-identity
+  namespace: kuma-system # only in system namespace
+  labels:
+    kuma.io/mesh: default
+spec:
+  selector:
+    dataplane:
+      matchLabels:
+        app: demo-app
+```
+
+The `MeshIdentity` resource’s selector field allows selection based on dataplane labels. Users can define it to apply to either all dataplanes or a specific subset.
+If there are multiple (X) identities selecting the same dataplane, the control plane will resolve the conflict by choosing one based on the following order of precedence:
+
+1. Selector specificity – the identity with most specific selector is preferred.
+2. Lexicographical order – if selector sizes are equal, the identity with the lexicographically smallest name is chosen.
+
+#### How would look the process of switching backends?
+
+1. User has a Mesh with MeshIdentity `identity-1`
+```yaml
+...
+name: demo-client-65f94cb577-jxrrq-44wfv42db9z44xxf
+type: DataplaneInsight
+mTLS:
+  certificateExpirationTime: "2025-06-03T17:35:33Z"
+  certificateRegenerations: 1
+  issuedBackend: system_kri_mi_mesh-1_us-east-2_kuma-system_identity-1
+  lastCertificateRegeneration: "2025-06-02T17:35:33.624189876Z"
+  supportedBackends: # list of MeshTrust name
+  - system_kri_mtrust_mesh-1_us-east-2_kuma-system_identity-1
+...
+```
+2. User creates MeshTrust with CA of a new MeshIdentity `identity-2`
+3. Trust is propagated. We provide identity and keep the information(supported CA (Trust), and Identity) in DataplaneInsight
+```yaml
+...
+name: demo-client-65f94cb577-jxrrq-44wfv42db9z44xxf
+type: DataplaneInsight
+mTLS:
+  certificateExpirationTime: "2025-06-03T17:35:33Z"
+  certificateRegenerations: 1
+  issuedBackend: system_kri_mi_mesh-1_us-east-2_kuma-system_identity-1 # name of MeshIdentity
+  lastCertificateRegeneration: "2025-06-02T17:35:33.624189876Z"
+  supportedBackends: # list of MeshTrust name
+  - system_kri_mtrust_mesh-1_us-east-2_kuma-system_identity-1
+  - system_kri_mtrust_mesh-1_us-east-2_kuma-system_identity-2
+...
+```
+
+Based on this information we will update MeshService
+
+```yaml
+...
+name: second-test-server-b7dwz26bb9b4z676
+spec:
+  identities:
+  - type: SpiffeID
+    value: spiffe://old.zone-1.0c5afd2e-de93-437b-8ba0-3decdff805b6.kuma.io/ns/real-resource-ns/sa/default
+  ports:
+  - appProtocol: http
+    name: main
+    port: 80
+    targetPort: main
+  selector:
+    dataplaneTags:
+      app: second-test-server
+      k8s.kuma.io/namespace: real-resource-ns
+  state: Available
+status:
+  ...
+  tls:
+    status: Ready <- only if all DPPs have identity provided
```

Comment:
Also I thought we don't want real traffic to be affected by the info we gathered through DataplaneInsights? But currently we use `status.tls.status` when configuring outbounds for MeshService

Reply:
We are using it now so I don't want to change this logic